bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java

```diff
@@ -424,4 +443,35 @@ public AutoSaveConfiguration getAutoSaveConfig() {
         return autoSaveConfig;
     }
 
+    @Override
+    public Map<String, Object> getSubmitProperties() {
+        Map<String, Object> submitProps = new LinkedHashMap<>();
+        Map<String, String> submitActionMap = new LinkedHashMap<>();
+
+        submitActionMap.put(FormConstants.RT_SUBMIT_TYPE_EMAIL, FormConstants.NAME_SUBMIT_TYPE_EMAIL);
+        submitActionMap.put(FormConstants.RT_SUBMIT_TYPE_SPREADSHEET, FormConstants.NAME_SUBMIT_TYPE_SPREADSHEET);
+        submitActionMap.put(FormConstants.RT_SUBMIT_TYPE_REST, FormConstants.NAME_SUBMIT_TYPE_REST);
+        submitActionMap.put(FormConstants.RT_SUBMIT_TYPE_FDM, FormConstants.NAME_SUBMIT_TYPE_FDM);
+        submitActionMap.put(FormConstants.RT_SUBMIT_TYPE_AZURE_BLOB, FormConstants.NAME_SUBMIT_TYPE_AZURE_BLOB);
+        submitActionMap.put(FormConstants.RT_SUBMIT_TYPE_ONEDRIVE, FormConstants.NAME_SUBMIT_TYPE_ONEDRIVE);
+        submitActionMap.put(FormConstants.RT_SUBMIT_TYPE_SHAREPOINT, FormConstants.NAME_SUBMIT_TYPE_SHAREPOINT);
+        submitActionMap.put(FormConstants.RT_SUBMIT_TYPE_AEM_WORKFLOW, FormConstants.NAME_SUBMIT_TYPE_AEM_WORKFLOW);
+        submitActionMap.put(FormConstants.RT_SUBMIT_TYPE_POWER_AUTOMATE, FormConstants.NAME_SUBMIT_TYPE_POWER_AUTOMATE);
+
+        submitProps.put("actionType", submitActionMap.getOrDefault(actionType, actionType));
+        if (mailto != null && mailto.length > 0) {
+            submitProps.put("email_mailto", mailto);
```

Comment:
Add reserved properties directly here and read the properties directly. Since we don't have getters, lets not inject them at top.